### PR TITLE
Fix the unix socket #ifdef in init.c

### PR DESCRIPTION
**Pull request by vjpai**
The ifdef was changed by #5755 but not set properly in init.c . This created a problem in environments with one but not the other.

### CODE REVIEW

**Comment by ctiller** [comment]
LGTM

**Comment by vjpai** [comment]
This should not merge yet. I have other issues to raise.

**Comment by vjpai** [comment]
Ok, this is ready to review. Thanks!

File: src/core/iomgr/unix_sockets_posix_noop.c [hunk 1]
```diff
@@ -35,7 +35,10 @@
 
 #ifndef GPR_HAVE_UNIX_SOCKET
 
-void grpc_create_socketpair_if_unix(int sv[2]) {}
+void grpc_create_socketpair_if_unix(int sv[2]) {
+  // TODO(ahedberg): Please fill in an implementation for non-Unix-sockets case
```
Comment (ahedberg) [inline src/core/iomgr/unix_sockets_posix_noop.c]: Please provide more direction when adding TODOs assigned to volunteer contributors (e.g. me) than "fill in an implementation". This case was previously unhandled IIUC.

Comment (vjpai) [inline src/core/iomgr/unix_sockets_posix_noop.c]: Yes, sorry about that; I've tried to be more explicit in #5946 but I am not sure that that's the right suggestion. I'll change the TODO here as well.

**Comment by vjpai** [comment]
Craig, does the LGTM still stand? I made a post-LGTM change.

**Comment by ctiller** [comment]
Still LGTM

**Pull request merged**